### PR TITLE
Added jest/unit_tests to verify the log parsing and processing flow

### PR DESCRIPTION
**Pull request by aayushchouhan09**
### Explain the changes
1. Added jest/unit_tests to verify the log parsing and processing flow and to support copy_object for log based replication. 

- [X] Tests added

### CODE REVIEW

File: src/native/chunk/splitter.cpp [hunk 1]
```diff
@@ -44,8 +44,21 @@ Splitter::Splitter(
     if (_calc_md5) {
         extern bool fips_mode;
         if (fips_mode) {
-            posix_memalign((void**)&_md5_mb_mgr, 16, sizeof(MD5_HASH_CTX_MGR));
-            posix_memalign((void**)&_md5_mb_ctx, 16, sizeof(MD5_HASH_CTX));
+            int result;
+            result = posix_memalign((void**)&_md5_mb_mgr, 16, sizeof(MD5_HASH_CTX_MGR));
+            if (result == 0) {
+                printf("Memory allocation successful. Aligned address: %p\n", _md5_mb_mgr);
```
Comment (dannyzaken) [inline src/native/chunk/splitter.cpp:50]: no need to print on success

File: src/native/chunk/splitter.cpp [hunk 2]
```diff
@@ -44,8 +44,21 @@ Splitter::Splitter(
...
+                free(ptr);
```
Comment (dannyzaken) [inline src/native/chunk/splitter.cpp:51]: why `free`?

Comment (achouhan09) [reply src/native/chunk/splitter.cpp:51]: Actually, its just a testing code to fix the CI job. I forgot to remove it.

File: src/native/third_party/libutp/utp_internal.cpp [hunk 3]
```diff
@@ -1975,7 +1975,7 @@ size_t utp_process_incoming(UTPSocket *conn, const byte *packet, size_t len, boo
 		if (pkt == 0 || pkt->transmissions == 0) continue;
 		assert((int)(pkt->payload) >= 0);
 		acked_bytes += pkt->payload;
-		if (conn->mtu_probe_seq && seq == conn->mtu_probe_seq) {
+		if (conn->mtu_probe_seq && seq == static_cast<int>(conn->mtu_probe_seq)) {
```
Comment (liranmauda) [inline src/native/third_party/libutp/utp_internal.cpp:1978]: same as isa-l, if this is an upstream project that we use, we first need to make the PR there

File: src/native/chunk/splitter.cpp [hunk 4]
```diff
@@ -44,8 +44,19 @@ Splitter::Splitter(
...
         if (fips_mode) {
-            posix_memalign((void**)&_md5_mb_mgr, 16, sizeof(MD5_HASH_CTX_MGR));
-            posix_memalign((void**)&_md5_mb_ctx, 16, sizeof(MD5_HASH_CTX));
+            int result;
+            result = posix_memalign((void**)&_md5_mb_mgr, 16, sizeof(MD5_HASH_CTX_MGR));
+            if (result == 0) {
+                // memory assigned successfully
+            } else {
+                // memory not assigned successfully, need to handle the case
+            }
```
Comment (liranmauda) [inline src/native/chunk/splitter.cpp:53]: Do we intentionally do an empty if...else?

File: src/native/third_party/isa-l/crc/crc_base.c [hunk 5]
```diff
@@ -267,7 +267,7 @@ static const uint32_t crc32_table_gzip_refl[256] = {
 
 uint16_t crc16_t10dif_base(uint16_t seed, uint8_t * buf, uint64_t len)
 {
-	int i;
+	uint64_t i;
```
[the remark this answers is not in the document]
Comment (achouhan09) [reply src/native/third_party/isa-l/crc/crc_base.c]: Thanks for raising the point. But these warnings are not creating error so we can remove these changes.

File: src/native/third_party/libutp/utp_internal.cpp (hunk 3 above)
Comment (achouhan09) [reply src/native/third_party/libutp/utp_internal.cpp]: Thanks, removing the changes.

File: src/native/chunk/splitter.cpp (hunk 1 above)
Comment (achouhan09) [reply src/native/chunk/splitter.cpp]: Thanks, updated the code.

File: src/native/chunk/splitter.cpp (hunk 4 above)
Comment (achouhan09) [reply src/native/chunk/splitter.cpp]: Thanks, applied some other fixes.

File: src/native/chunk/splitter.cpp [hunk 6]
```diff
@@ -44,8 +44,9 @@ Splitter::Splitter(
     if (_calc_md5) {
         extern bool fips_mode;
         if (fips_mode) {
-            posix_memalign((void**)&_md5_mb_mgr, 16, sizeof(MD5_HASH_CTX_MGR));
-            posix_memalign((void**)&_md5_mb_ctx, 16, sizeof(MD5_HASH_CTX));
+            [[maybe_unused]] int result;
+            result = posix_memalign((void**)&_md5_mb_mgr, 16, sizeof(MD5_HASH_CTX_MGR));
+            result = posix_memalign((void**)&_md5_mb_ctx, 16, sizeof(MD5_HASH_CTX));
```
Comment (liranmauda) [inline src/native/chunk/splitter.cpp:49]: why do we set it into a variable (that we are not using?)

Comment (achouhan09) [reply src/native/chunk/splitter.cpp:49]: We need to handle the return value of `posix_memalign()` if don't do that than it gives a warning which is being treated as an error on CI job.

**Review by liranmauda — APPROVED** [review]

**Review by dannyzaken — APPROVED** [review]

**Pull request merged**